pkg/roachpb/span_config.proto

```diff
@@ -140,6 +141,11 @@ message SpanConfig {
   // preferred option to least. The first preference that an existing replica of
   // a range matches will take priority for the lease.
   repeated LeasePreference lease_preferences = 9 [(gogoproto.nullable) = false];
+
+  // ProtectedTimestamps captures all the protected timestamp records that apply
+  // to the range. The timestamp values represent the timestamps after which
+  // data will be protected from GC, if the protection succeeds.
+  repeated util.hlc.Timestamp protected_timestamps = 10 [(gogoproto.nullable) = false];
```

Comment:
this is subject to change in https://github.com/cockroachdb/cockroach/pull/74765